modules/apps/knowledge-base/knowledge-base-service/src/main/java/com/liferay/knowledge/base/service/impl/KBArticleLocalServiceImpl.java

```diff
@@ -1563,6 +1589,92 @@ private void _deleteSubscriptions(KBArticle kbArticle)
 		}
 	}
 
+	private void _expireKBArticlesByCompanyId(
+			long companyId, Date expirationDate, ServiceContext serviceContext)
+		throws PortalException {
+
+		long userId = _getActiveCompanyAdminUserId(companyId);
```

Comment:
Why are you picking the first active user? This is a system process, so it makes sense to use the "system" user (the default user for the company). Otherwise, users may be confused, seeing the portal claims they modified something they really didn't.

Reply:
Do you think it's good idea to change this in others components that changes this this way?

Reply:
Not for the moment. I may be wrong :-)